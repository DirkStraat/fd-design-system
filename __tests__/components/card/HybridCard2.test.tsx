import React from 'react';
import {
    HybridCard2,
    getCssClassNames,
} from '../../../src/components/card/HybridCard2';
import { render } from '@testing-library/react';

describe('HybridCard2', () => {
    test('should render correctly', () => {
        const { container, getByText } = render(
            <HybridCard2
                id="1"
                intro="This is the intro"
                label="Ondernemen"
                title="This is the title"
                url="https://example.com/1/This is the title"
            />
        );

        expect(getByText('This is the title')).toBeTruthy();
        expect(getByText('Ondernemen')).toBeTruthy();
        expect(getByText('This is the intro')).toBeTruthy();
        expect(container.querySelector('[href]').getAttribute('href')).toBe(
            'https://example.com/1/This is the title'
        );
    });

    test('should render is-read correctly', () => {
        const { container } = render(
            <HybridCard2
                id="1"
                intro="This is the intro"
                label="Ondernemen"
                title="This is the title"
                url="https://example.com/1/This is the title"
                isRead={true}
            />
        );
        expect(container).toMatchSnapshot();
    });

    test('should render variant-1 correctly', () => {
        const { container } = render(
            <HybridCard2
                id="1"
                intro="This is the intro"
                label="Ondernemen"
                title="This is the title"
                url="https://example.com/1/This is the title"
                variant="variant-1"
            />
        );
        expect(container).toMatchSnapshot();
    });

    test('should have correct class name', () => {
        const { container } = render(
            <HybridCard2
                id="1"
                intro="This is the intro"
                label="Ondernemen"
                title="This is the title"
                url="https://example.com/1/This is the title"
            />
        );

        expect(
            container.firstElementChild.classList.contains(
                getCssClassNames()[0]
            )
        ).toBeTruthy();
    });
});
