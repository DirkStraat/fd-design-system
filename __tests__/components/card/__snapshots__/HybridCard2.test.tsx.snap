// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HybridCard2 should render is-read correctly 1`] = `
<div>
  <article
    class="hybridCard2 isRead"
    id="1"
  >
    <a
      aria-label="This is the title"
      href="https://example.com/1/This is the title"
    >
      <div
        class="teaserText"
      >
        <div
          class="meta"
        >
          <span
            class="prefix"
          >
            Ondernemen
          </span>
        </div>
        <div
          class="textContainer"
        >
          <h1
            class="heading serif s"
          >
            This is the title
          </h1>
          <p
            class="intro body-text sans s"
          >
            This is the intro
          </p>
        </div>
      </div>
    </a>
    <footer
      class="footer footer"
    >
      <menu>
        <a>
          <span
            class="icon"
          />
          <span
            data-bookmark="bookmark"
          >
            Bewaren
          </span>
        </a>
      </menu>
    </footer>
  </article>
</div>
`;

exports[`HybridCard2 should render variant-1 correctly 1`] = `
<div>
  <article
    class="hybridCard2"
    data-variant="variant-1"
    id="1"
  >
    <a
      aria-label="This is the title"
      href="https://example.com/1/This is the title"
    >
      <div
        class="teaserText"
      >
        <div
          class="meta"
        >
          <span
            class="prefix"
          >
            Ondernemen
          </span>
        </div>
        <div
          class="textContainer"
        >
          <h1
            class="heading serif s"
          >
            This is the title
          </h1>
          <p
            class="intro body-text sans s"
          >
            This is the intro
          </p>
        </div>
      </div>
    </a>
    <footer
      class="footer footer"
    >
      <menu>
        <a>
          <span
            class="icon"
          />
          <span
            data-bookmark="bookmark"
          >
            Bewaren
          </span>
        </a>
      </menu>
    </footer>
  </article>
</div>
`;
